packages/studio-web/src/app/upload/upload.component.ts

```diff
@@ -87,6 +75,16 @@ export class UploadComponent implements OnInit {
         $localize`Whoops, something went wrong while recording!`
       );
     });
+    this.rasService.getLangs$().subscribe({
+      next: (langs: Array<SupportedLanguage>) => {
+        this.langs = langs.sort((a, b) => {
+          if (a.names['_'] < b.names['_']) return -1;
+          if (a.names['_'] > b.names['_']) return 1;
+          return 0;
+        });
+      },
+      error: (err) => this.reportRasError(err),
+    });
```

Comment:
I have to agree this code is less confusing that what we had before, thanks for cleaning up language sorting. (The sorting part was mine, and I have to admit I was just baffled by how hard it was to write `sorted(my_array, key=lambda x: x.names['_'])` in JS/TS.

I still have a question, though: I thought we were supposed to avoid doing real work like this in constructors, and defer it to `ngOnInit`, so that instantiation of the component for unit testing would not trigger these calls, only actual component initialization would. We were already initializing at construction time before you changes, so my question is orthogonal to this PR, but shouldn't we move this to `ngOnInit()`?

Reply:
That's a really good question, maybe @roedoejet knows the answer? We need `langs` to be defined before the component gets rendered... so, if `ngOnInit()` happens before rendering, we probably should do it there.

Reply:
also - it turns out this exists, for string comparison, so perhaps we should use it: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/localeCompare

sorry I called your code gross! it's not you, it's JavaScript :)

Reply:
My reading of this https://angular.io/guide/lifecycle-hooks is that yes, `ngOnInit` is the right place for it, I will make a PR with that change and the other one I just mentioned.

Reply:
nope. Angular is aggressive about rendering, so if we load the langs in `ngOnInit` we get a Flash Of Unintented Content with "Server Error" before it finishes loading them.

Reply:
Yeah, I saw `localeCompare` when I searched, but I didn't like adding a dependency just for this.

And good to know it has to go in the constructor, thanks!

Reply:
Oh, wait, I think it was because `localeCompare` sorts an array of strings, but I needed to dig into the structure for the key, which Python supports trivially with a lambda function, but apparently not JS.

Anyway, yes, JS is confusing. Often!